theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer-toolbar-contribution.tsx

```diff
@@ -114,28 +146,57 @@ export class TraceViewerToolbarContribution extends AbstractViewContribution<Wid
                     });
                 }}></div>
             </div>,
-            priority: 1,
+            priority: 4,
             group: 'navigation',
             onDidChange: this.onAnnotationsChangedEvent,
         });
-
-        registry.registerItem({
-            id: TraceViewerToolbarCommands.ZOOM_IN.id,
-            command: TraceViewerToolbarCommands.ZOOM_IN.id,
-            tooltip: TraceViewerToolbarCommands.ZOOM_IN.label,
-            priority: 1,
-        });
-        registry.registerItem({
-            id: TraceViewerToolbarCommands.ZOOM_OUT.id,
-            command: TraceViewerToolbarCommands.ZOOM_OUT.id,
-            tooltip: TraceViewerToolbarCommands.ZOOM_OUT.label,
-            priority: 2,
-        });
         registry.registerItem({
-            id: TraceViewerToolbarCommands.RESET.id,
-            command: TraceViewerToolbarCommands.RESET.id,
-            tooltip: TraceViewerToolbarCommands.RESET.label,
-            priority: 3,
+            id: TraceViewerToolbarCommands.MARKERS.id,
+            isVisible: w => w instanceof TraceViewerWidget && this.markersSetMap.size > 0,
+            // render() here is not a react component and hence need to disable the react display-name rule
+            // eslint-disable-next-line react/display-name
+            render: () => <div className="item enabled">
+                <div id="trace.viewer.toolbar.filter" className="fa fa-bars" title="Markers filter" onClick={async (event: React.MouseEvent) => {
```

Comment:
This hamburger menu has the same title as the filter menu.

Not sure if this should be a sub-menu in the hamburger menu instead? And what should be the tooltip of the hamburger menu?

Reply:
I changed the title to 'Marker Sets'. I didn't understand the sub-menu part.